src/Integration/Laravel/Queue/tests/Unit/AsyncAwsSqsQueueTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\Illuminate\Queue\Tests\Unit;
+
+use AsyncAws\Core\Test\ResultMockFactory;
+use AsyncAws\Illuminate\Queue\AsyncAwsSqsQueue;
+use AsyncAws\Illuminate\Queue\Job\AsyncAwsSqsJob;
+use AsyncAws\Sqs\Enum\MessageSystemAttributeName;
+use AsyncAws\Sqs\Enum\QueueAttributeName;
+use AsyncAws\Sqs\Result\GetQueueAttributesResult;
+use AsyncAws\Sqs\Result\ReceiveMessageResult;
+use AsyncAws\Sqs\Result\SendMessageResult;
+use AsyncAws\Sqs\SqsClient;
+use AsyncAws\Sqs\ValueObject\Message;
+use Illuminate\Container\Container;
+use PHPUnit\Framework\TestCase;
+
+class AsyncAwsSqsQueueTest extends TestCase
+{
+    private $queue = 'https://sqs.us-east-2.amazonaws.com/123456789012/invoice';
+
```

Comment:
We need to test the most common use case in Laravel which is to push a job class. 

Queue::push(new CreateUser); 

Because Laravel serializes the job and brings it back on the pull.

Reply:
I extend `Illuminate\Queue\Queue` so the serialisation process is not done in `AsyncAwsSqsQueue`. But a test for this is good to make sure we don't override this in the future.

Reply:
I've added a test for that now. 

Thank you